xarray/core/indexing.py

```diff
@@ -1397,17 +1397,17 @@ def __init__(self, array: Any, dtype: DTypeLike = None):
         self.array = utils.safe_cast_to_index(array)
         if dtype is None:
             if isinstance(array, pd.PeriodIndex):
-                dtype = np.dtype("O")
+                dtype_ = np.dtype("O")
```

Comment:
This could potentially be simplified by [allowing-redefinitions](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-allow-redefinition)